src/main/java/picard/fingerprint/CheckFingerprint.java

```diff
@@ -129,37 +134,65 @@ protected int doWork() {
+        List<FingerprintResults> results;
 
-        SequenceUtil.assertSequenceDictionariesEqual(SAMSequenceDictionaryExtractor.extractDictionary(INPUT), SAMSequenceDictionaryExtractor.extractDictionary(GENOTYPES), true);
-        SequenceUtil.assertSequenceDictionariesEqual(SAMSequenceDictionaryExtractor.extractDictionary(INPUT), checker.getHeader().getSequenceDictionary(), true);
+        String observedSampleAlias = null;
+        final boolean isBamOrSamFile = isBamOrSamFile(INPUT);
+        if (isBamOrSamFile) {
+            SequenceUtil.assertSequenceDictionariesEqual(SAMSequenceDictionaryExtractor.extractDictionary(INPUT), SAMSequenceDictionaryExtractor.extractDictionary(GENOTYPES), true);
+            SequenceUtil.assertSequenceDictionariesEqual(SAMSequenceDictionaryExtractor.extractDictionary(INPUT), checker.getHeader().getSequenceDictionary(), true);
 
-        // If sample alias isn't supplied, assume it's the one from the INPUT file's RGs
-        if (SAMPLE_ALIAS == null) {
+            // Verify that there's only one sample in the SAM/BAM.
             final SamReader in = SamReaderFactory.makeDefault().open(INPUT);
             for (final SAMReadGroupRecord rec : in.getFileHeader().getReadGroups()) {
-                if (SAMPLE_ALIAS == null) {
-                    SAMPLE_ALIAS = rec.getSample();
+                if (observedSampleAlias == null) {
+                    observedSampleAlias = rec.getSample();
                 }
-                else if (!SAMPLE_ALIAS.equals(rec.getSample())) {
-                    throw new PicardException("SAM File must not contain data from multiple samples.");
+                else if (!observedSampleAlias.equals(rec.getSample())) {
+                    throw new PicardException("INPUT SAM/BAM file must not contain data from multiple samples.");
                 }
             }
             CloserUtil.close(in);
-        }
 
+            // If expected sample alias isn't supplied, assume it's the one from the INPUT file's RGs
+            if (EXPECTED_SAMPLE_ALIAS == null) {
+                EXPECTED_SAMPLE_ALIAS = observedSampleAlias;
+            }
+
+            results = checker.checkFingerprints(
+                    Collections.singletonList(INPUT),
+                    Collections.singletonList(GENOTYPES),
+                    EXPECTED_SAMPLE_ALIAS,
+                    IGNORE_READ_GROUPS);
+        } else {            // Input is a VCF
+            // Note that FingerprintChecker.loadFingerprints() verifies that the VCF's Sequence Dictionaries agree with that of the Haplotye Map File
+
+            // Verify that there is only one sample in the VCF
+            final VCFFileReader fileReader = new VCFFileReader(INPUT, false);
+            final VCFHeader fileHeader = fileReader.getFileHeader();
+            if (fileHeader.getNGenotypeSamples() != 1) {
+                throw new PicardException("INPUT VCF file must not contain data from multiple samples.");
```

Comment:
a vcf could also have getNGenotypeSamples()==0...so your error message will be incorrect.